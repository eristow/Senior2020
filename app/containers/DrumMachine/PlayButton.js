import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import { useInjectReducer } from 'utils/injectReducer';
import { makeSelectPlaying } from './selectors';
import { togglePlay } from './actions';
import reducer from './reducer';

const Play = styled.button`
  border: 2px solid ${props => (props.on === false ? '#0BA134' : '#FD7272')};
  border: 2px solid #ddd;
  color: #fff;
  background: ${props => (props.on === false ? '#0BA134' : '#FD7272')};
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  padding: 10px;
  font-size: 18px;
  border-radius: 4px;
  margin: 2px 4px;
  margin-left: 6px;
  align-self: center;
  min-width: 100px;

  &:active {
    color: ${props => (props.on === false ? '#0BA134' : '#FD7272')};
    background: #666666;
  }
`;

const key = 'drumMachine';

export function PlayButton({ playing, onClickPlay }) {
  useInjectReducer({ key, reducer });
  return (
    <Play on={playing} onClick={onClickPlay}>
      {playing ? 'Stop' : 'Play'}
    </Play>
  );
}

PlayButton.propTypes = {
  playing: PropTypes.bool,
  onClickPlay: PropTypes.func,
};

const mapStateToProps = createStructuredSelector({
  playing: makeSelectPlaying(),
});

const mapDispatchToProps = dispatch => ({
  onClickPlay: evt => {
    dispatch(togglePlay(evt.target.value));
  },
});

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(withConnect)(PlayButton);
