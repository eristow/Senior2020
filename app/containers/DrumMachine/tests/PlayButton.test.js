import React from 'react';
import { render } from 'react-testing-library';
import { Provider } from 'react-redux';
import configureStore from '../../../configureStore';

import { PlayButton } from '../PlayButton';

describe('<PlayButton />', () => {
  let store;

  beforeAll(() => {
    store = configureStore();
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();

    render(
      <Provider store={store}>
        <PlayButton dispatch={dispatch} />
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <PlayButton />
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
