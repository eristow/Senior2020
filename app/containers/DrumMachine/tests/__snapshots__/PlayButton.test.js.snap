// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayButton /> Should render and match the snapshot 1`] = `
<button
  class="PlayButton__Play-sc-1sbrdge-0 kGFvTg"
>
  Play
</button>
`;
